### PR TITLE
Improve ci and get working with win32.

### PR DESCRIPTION
**Pull request by richardkiss**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,123 @@
+name: Build
+
+on: [push, pull_request]
```
Comment (arvidn) [inline .github/workflows/build.yml:3]: this will make all the test run twice when you update a pull request

Comment (richardkiss) [reply .github/workflows/build.yml:3]: I just copied this part from elsewhere, I don't have much of an opinion about it.

**Comment by arvidn** [comment]
looks good. Is there a good reason to build release builds? `maturin develop --release`

**Comment by richardkiss** [comment]
That's what I've been doing for local development. Presumably the builds are slower and the tests are faster. Maybe at some point we should have some benchmark testing of lengthy programs which gets run separately so we can eyeball if changes make things slower.

**Comment by arvidn** [comment]
my concern is that if there are any `debug_assert!()`s, they won't fire in release builds.

File: .github/workflows/build.yml (hunk 1 above)
Comment (arvidn) [reply .github/workflows/build.yml:3]: I think this is the only important comment. As you can see in this PR, all tests ran twice.

**Comment by richardkiss** [comment]
Squash merged separately.

**Pull request closed without merging**